recipe_scrapers/joshuaweissman.py

```diff
@@ -21,10 +21,29 @@ def category(self):
         return self.schema.category()
 
     def total_time(self):
-        raise NotImplementedError()
+        total = self.soup.find(
+            lambda tag: tag.name == "span" and "Total Time" in tag.get_text()
+        )
```

Comment:
Nice, thanks for implementing this.

Could the `find` call be modified to pre-filter to `span` elements, and then to use a `lambda` to find the text?

I'm not sure whether that'd make a performance difference in practice, but it could do.

Reply:
Makes sense, I'll push that in a sec

Reply:
ok, probably spent too long on that. only iterates over spans once, but a bit less legible

Reply:
Mm, and after seeing how many `span`s there are on the page, perhaps this wasn't a useful suggestion by me.  Thanks for exploring it though.  It might be possible to improve the legibility - are there any sections of the code you think are particularly tricky?  (the inline method seems clever to me, due to the re-use of the outside-scoped `time_texts` variable)

Reply:
All good! Was interesting to try and get them to run through them only once. Overall I still think it is quite legible, just only slightly less than before. Thanks for your help!

Reply:
You're welcome - thanks for investigating it!